hydra-cluster/exe/hydra-cluster/Main.hs

```diff
@@ -24,13 +26,21 @@ run options =
     withStateDirectory $ \workDir ->
       case knownNetwork of
         Just network -> do
-          when (useMithril == UseMithril) $
-            downloadLatestSnapshotTo (contramap FromMithril tracer) network workDir
-          withCardanoNodeOnKnownNetwork fromCardanoNode workDir network $ \node -> do
-            waitForFullySynchronized fromCardanoNode node
-            publishOrReuseHydraScripts tracer node
-              >>= singlePartyHeadFullLifeCycle tracer workDir node
-        Nothing ->
+          maybeRunningNode <- findRunningCardanoNode workDir
+          case maybeRunningNode of
+            Nothing -> do
+              when (useMithril == UseMithril) $
+                downloadLatestSnapshotTo (contramap FromMithril tracer) network workDir
+              withNewCardanoNodeOnKnownNetwork fromCardanoNode workDir network $ \node -> do
+                waitForFullySynchronized fromCardanoNode node
+                publishOrReuseHydraScripts tracer node
+                  >>= singlePartyHeadFullLifeCycle tracer workDir node
+            Just node -> do
+              waitForFullySynchronized fromCardanoNode node
+              publishOrReuseHydraScripts tracer node
+                >>= singlePartyHeadFullLifeCycle tracer workDir node
+        Nothing -> do
+          removeDirectoryRecursive $ workDir </> "db"
```

Comment:
Why is this removing of `db` needed now?

Reply:
because it was removed from the github action

Reply:
because it was removed from the smoke test workflow.

I'll move the moving this operation to above as requested [here](https://github.com/input-output-hk/hydra/pull/1285#discussion_r1480038804)